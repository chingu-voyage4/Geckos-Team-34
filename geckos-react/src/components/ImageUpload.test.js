import React from 'react';
import { shallow, mount } from 'enzyme';

import ImageUpload from './ImageUpload';

describe('<ImageUpload />', () => {
  let wrapper;
  const newFile = new File(['file contents'], "filename");

  beforeEach(() => {
    wrapper = shallow(<ImageUpload imgFile={jest.fn()} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  it('should render a file input and update the components state with file info', () => {
    const elem = wrapper.find('#file');

    elem.simulate('change', { target: { name: 'file', files: 'test.jpg' } });

    expect(elem.length).toBe(1);
    expect(wrapper.state().file).toEqual('test.jpg');
  });

  it('should render a button input', () => {
    const elem = wrapper.find('button');
    expect(elem.length).toBe(1);
  });

  it('should call onImageChange', () => {

  });

  it('should call sendFileInfo', () => {
    const spy = jest.fn();
    wrapper = shallow(<ImageUpload imgFile={spy} />);
    wrapper.find('button').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});

