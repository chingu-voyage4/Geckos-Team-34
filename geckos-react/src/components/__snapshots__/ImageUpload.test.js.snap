// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUpload /> should render correctly 1`] = `
<FormField
  width={4}
>
  <input
    accept=".png, .jpg, .jpeg"
    id="file"
    name="file"
    onChange={[Function]}
    type="file"
  />
  <button
    className="ui button"
    onClick={[Function]}
  >
    <i
      className="upload icon"
    />
    Upload
  </button>
</FormField>
`;
